src/SFA.DAS.ApprenticeFeedback.Application/Commands/CreateApprenticeFeedback/CreateApprenticeFeedbackHandler.cs

```diff
@@ -33,15 +34,45 @@ public async Task<CreateApprenticeFeedbackResponse> Handle(CreateApprenticeFeedb
             }
 
             var validAttributes = await _apprenticeFeedbackRepository.GetAttributes();
+            var validAttributesIds = validAttributes.Select(a => a.AttributeId).ToList();
+            var providedAttributesIds = request.FeedbackAttributes.Select(a => a.Id).ToList();
+            var invalidAttributesIds = providedAttributesIds.Except(validAttributesIds).ToList();
+            var attributesProvidedNames = GetAttributeNames(request);
+            string invalidAttributeNamesOutput = (invalidAttributesIds.Count != 0) ? CreateInvalidOutput(request, invalidAttributesIds) : string.Empty;
 
             var allValidAttributes = request.FeedbackAttributes.Select(s => s.Id).All(t => validAttributes.Select(t => t.AttributeId).Contains(t));
             if (!allValidAttributes)
             {
-                _logger.LogError($"Some or all of the attributes supplied to create the feedback record do not exist in the database. Attributes provided in the request: {request.FeedbackAttributes.ToList()}");
-                throw new InvalidOperationException($"Some or all of the attributes supplied to create the feedback record do not exist in the database. Attributes provided in the request: {request.FeedbackAttributes.ToList()}");
+                _logger.LogError($"Some or all of the attributes supplied to create the feedback record do not exist in the database. Attributes provided in the request: {attributesProvidedNames}, the following attributes are invalid: {invalidAttributeNamesOutput}");
+                throw new InvalidOperationException($"Some or all of the attributes supplied to create the feedback record do not exist in the database. Attributes provided in the request: {attributesProvidedNames}, the following attributes are invalid: {invalidAttributeNamesOutput}");
             }
 
-            var feedback = new Domain.Entities.ApprenticeFeedbackResult
+            var feedback = CreateApprenticeFeedbackResult(request, apprenticeFeedbackTarget);
+
+            feedback = await _apprenticeFeedbackRepository.CreateApprenticeFeedbackResult(feedback);
+
+            _logger.LogDebug($"Successfully created feedback object with Id: {feedback.Id}");
+            return new CreateApprenticeFeedbackResponse() { ApprenticeFeedbackResultId = feedback.Id };
+        }
+
+        private string GetAttributeNames(CreateApprenticeFeedbackCommand request)
+        {
+            var attributeNames = request.FeedbackAttributes.Select(a => a.Name).ToList();
+            return string.Join(", ", attributeNames);
+        }
+
+
+        private string CreateInvalidOutput(CreateApprenticeFeedbackCommand request, List<int> invalidAttributesIds)
+        {
+            var invalidAttributesNames = (from a in request.FeedbackAttributes
```

Comment:
This is perfectly valid Linq to achieve the goal here, as its using query syntax over method syntax, but if you wanted to use method syntax it can be done

`request.FeedbackAttributes.Where(s => invalidAttributeIds.Contains(s.Id)).Select(t => t.Name)`